Failing main branch — test_save_mfdataset_compute_false_roundtrip

<!-- Please include a self-contained copy-pastable example that generates the issue if possible.

Please be concise with code posted. See guidelines below on how to provide a good bug report:

- Craft Minimal Bug Reports: http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports
- Minimal Complete Verifiable Examples: https://stackoverflow.com/help/mcve

Bug reports that follow these guidelines are easier to diagnose, and so are often handled much more quickly.
-->

We had the main branch passing for a while, but unfortunately another test failure. Now in our new `Linux py38-backend-api-v2` test case, in`test_save_mfdataset_compute_false_roundtrip`

[link](https://dev.azure.com/xarray/xarray/_build/results?buildId=4089&view=ms.vss-test-web.build-test-results-tab&runId=63370&resultId=101311&paneView=debug)

```
self = <xarray.tests.test_backends.TestDask object at 0x7f821a0d6190>

    def test_save_mfdataset_compute_false_roundtrip(self):
        from dask.delayed import Delayed
    
        original = Dataset({"foo": ("x", np.random.randn(10))}).chunk()
        datasets = [original.isel(x=slice(5)), original.isel(x=slice(5, 10))]
        with create_tmp_file(allow_cleanup_failure=ON_WINDOWS) as tmp1:
            with create_tmp_file(allow_cleanup_failure=ON_WINDOWS) as tmp2:
                delayed_obj = save_mfdataset(
                    datasets, [tmp1, tmp2], engine=self.engine, compute=False
                )
                assert isinstance(delayed_obj, Delayed)
                delayed_obj.compute()
                with open_mfdataset(
                    [tmp1, tmp2], combine="nested", concat_dim="x"
                ) as actual:
>                   assert_identical(actual, original)
E                   AssertionError: Left and right Dataset objects are not identical
E                   
E                   
E                   Differing data variables:
E                   L   foo      (x) float64 dask.array<chunksize=(5,), meta=np.ndarray>
E                   R   foo      (x) float64 dask.array<chunksize=(10,), meta=np.ndarray>

/home/vsts/work/1/s/xarray/tests/test_backends.py:3274: AssertionError

AssertionError: Left and right Dataset objects are not identical

Differing data variables:
L   foo      (x) float64 dask.array<chunksize=(5,), meta=np.ndarray>
R   foo      (x) float64 dask.array<chunksize=(10,), meta=np.ndarray>
```

@aurghs & @alexamici — are you familiar with this? Thanks in advance

--- a/xarray/tests/test_distributed.py
+++ b/xarray/tests/test_distributed.py
@@ -144,6 +144,10 @@ def test_open_mfdataset_can_open_files_with_cftime_index(parallel, tmp_path):
 @requires_netCDF4
 @pytest.mark.parametrize("parallel", (True, False))
 def test_open_mfdataset_multiple_files_parallel_distributed(parallel, tmp_path):
+    if parallel:
+        pytest.skip(
+            "Flaky in CI. Would be a welcome contribution to make a similar test reliable."
+        )
     lon = np.arange(100)
     time = xr.cftime_range("20010101", periods=100, calendar="360_day")
     data = np.random.random((time.size, lon.size))

--- a/xarray/tests/test_distributed.py
+++ b/xarray/tests/test_distributed.py
@@ -144,10 +144,6 @@ def test_open_mfdataset_can_open_files_with_cftime_index(parallel, tmp_path):
 @requires_netCDF4
 @pytest.mark.parametrize("parallel", (True, False))
 def test_open_mfdataset_multiple_files_parallel_distributed(parallel, tmp_path):
-    if parallel:
-        pytest.skip(
-            "Flaky in CI. Would be a welcome contribution to make a similar test reliable."
-        )
     lon = np.arange(100)
     time = xr.cftime_range("20010101", periods=100, calendar="360_day")
     data = np.random.random((time.size, lon.size))
@@ -172,6 +168,10 @@ def test_open_mfdataset_multiple_files_parallel_distributed(parallel, tmp_path):
 @requires_netCDF4
 @pytest.mark.parametrize("parallel", (True, False))
 def test_open_mfdataset_multiple_files_parallel(parallel, tmp_path):
+    if parallel:
+        pytest.skip(
+            "Flaky in CI. Would be a welcome contribution to make a similar test reliable."
+        )
     lon = np.arange(100)
     time = xr.cftime_range("20010101", periods=100, calendar="360_day")
     data = np.random.random((time.size, lon.size))

